internal/app/app.go

```diff
@@ -46,38 +52,53 @@ func (a *App) CreateShortURL(res http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	short := shorten.Shorting(string(body), a.storage)
+	short, err := service.ServiceCreate(string(body), a.cfg.DefURL, a.storage)
+	if err != nil {
+		logger.Initialize().Info(err)
+		res.Header().Set("content-type", "text/plain; charset=utf-8")
+		res.WriteHeader(http.StatusConflict)
+		res.Write([]byte(short))
+		return
+	}
 
 	res.Header().Set("content-type", "text/plain; charset=utf-8")
 
 	res.WriteHeader(http.StatusCreated)
 
-	fmt.Fprint(res, a.cfg.DefURL, "/", short)
+	res.Write([]byte(short))
 
 }
 
-type URL struct {
-	LongURL  string `json:"url"`
-	ShortURL string `json:"result"`
-}
-
 func (a *App) CreateJSONShorten(res http.ResponseWriter, req *http.Request) {
 	body, err := io.ReadAll(req.Body)
 	if err != nil {
 		return
 	}
 
-	var url URL
+	var url models.URL
 
 	err = json.Unmarshal(body, &url)
 	if err != nil {
 		logger.Initialize().Info(err)
 	}
 
-	shortURL := shorten.Shorting(url.LongURL, a.storage)
-	shortenMap := make(map[string]string)
+	shortURL, err := service.ServiceCreate(url.LongURL, a.cfg.DefURL, a.storage)
+	if err != nil {
+		shortenMap := make(map[string]string)
+
+		shortenMap["result"] = shortURL
 
-	shortURL = a.cfg.DefURL + "/" + shortURL
+		marshal, err2 := json.Marshal(shortenMap)
```

Comment:
```suggestion
		marshal, err := json.Marshal(shortenMap)
```
редко когда нужны две ошибки. Обработали - если ничего ничего критичного, записываем в эту же переменную новую ошибку